felix/bpf/maps.go

```diff
@@ -314,13 +343,50 @@ func (b *PinnedMap) Open() error {
 	return err
 }
 
+func (b *PinnedMap) compareSize() (bool, error) {
+	// Get current map info
+	mapInfo, err := GetMapInfo(b.fd)
+	if err != nil {
+		return false, fmt.Errorf("error getting map info %s", b.versionedName())
+	}
+	// If the current map size and the configured size are the same, nothing needs to
+	// be done.
+	if mapInfo.MaxEntries == b.MaxEntries {
+		return true, nil
+	}
+	// If the config is different, repin the existing map "path_old" and remove
+	// the current pinned path.
+	err = RepinMap(b.versionedName(), b.Path()+"_old")
+	if err != nil {
+		return false, fmt.Errorf("error repinning %s to %s: %w", b.Path(), b.Path()+"_old", err)
+	}
+	err = os.Remove(b.Path())
+	if err != nil {
+		return false, fmt.Errorf("error removing the pin %s", b.versionedName())
+	}
+	return false, nil
+}
+
 func (b *PinnedMap) EnsureExists() error {
+	var oldfd MapFD
+	sameSize := true
 	if b.fdLoaded {
 		return nil
 	}
 
 	if err := b.Open(); err == nil {
-		return nil
+		// store the old fd
+		// nolint
+		oldfd = b.MapFD()
```

Comment:
is the oldfd closed (does it need to be?) if we return because of an error or same size? What about using a defer here? Would it remove the need for `nolint`?